2D parallel_for_work_group fails to compile

**Bug summary**
parallel_for_work_group with 2d ranges fails to compile.

**To Reproduce**
Attempt to compile the code below:

```cpp
#include <SYCL/sycl.hpp>

int main()
{
    sycl::range<2> wg_size{16, 16};
    sycl::range<2> n_wgs{1, 1};
            
    sycl::queue q;
    q.submit([&](sycl::handler& cgh) {
        cgh.parallel_for_work_group<class wg>(n_wgs, wg_size,
            [=](sycl::group<2> group) {
                group.parallel_for_work_item([&](sycl::h_item<2> item) {
                    // do nothing
                });
            });
    });
}
```

**Expected behavior**
Compiles

**Actual behavior**
Compile error:
```
In file included from /opt/hipSYCL/include/hipSYCL/glue/kernel_launcher_factory.hpp:46,
                 from /opt/hipSYCL/include/hipSYCL/sycl/handler.hpp:61,
                 from /opt/hipSYCL/include/hipSYCL/sycl/queue.hpp:47,
                 from /opt/hipSYCL/include/hipSYCL/sycl/sycl.hpp:74,
                 from /opt/hipSYCL/include/SYCL/sycl.hpp:31,
                 from /home/park/example/example.cpp:1:
/opt/hipSYCL/include/hipSYCL/glue/omp/omp_kernel_launcher.hpp: In instantiation of ‘hipsycl::glue::omp_dispatch::parallel_for_workgroup(Function, hipsycl::sycl::range<dimensions>, hipsycl::sycl::range<dimensions>, size_t, Reductions ...) [with int Dim = 2; Function = main()::<lambda(hipsycl::sycl::handler&)>::<lambda(hipsycl::sycl::group<2>)>; Reductions = {}; size_t = long unsigned int]::<lambda(auto:7& ...)> [with auto:7 = {}]’:
/usr/include/c++/8/type_traits:2246:26:   required by substitution of ‘template<class _Fn, class ... _Args> static std::__result_of_success<decltype (declval<_Fn>()((declval<_Args>)()...)), std::__invoke_other> std::__result_of_other_impl::_S_test(int) [with _Fn = hipsycl::glue::omp_dispatch::parallel_for_workgroup(Function, hipsycl::sycl::range<dimensions>, hipsycl::sycl::range<dimensions>, size_t, Reductions ...) [with int Dim = 2; Function = main()::<lambda(hipsycl::sycl::handler&)>::<lambda(hipsycl::sycl::group<2>)>; Reductions = {}; size_t = long unsigned int]::<lambda(auto:7& ...)>&; _Args = {}]’
/usr/include/c++/8/type_traits:2257:55:   required from ‘struct std::__result_of_impl<false, false, hipsycl::glue::omp_dispatch::parallel_for_workgroup(Function, hipsycl::sycl::range<dimensions>, hipsycl::sycl::range<dimensions>, size_t, Reductions ...) [with int Dim = 2; Function = main()::<lambda(hipsycl::sycl::handler&)>::<lambda(hipsycl::sycl::group<2>)>; Reductions = {}; size_t = long unsigned int]::<lambda(auto:7& ...)>&>’
/usr/include/c++/8/type_traits:2262:12:   required from ‘struct std::__invoke_result<hipsycl::glue::omp_dispatch::parallel_for_workgroup(Function, hipsycl::sycl::range<dimensions>, hipsycl::sycl::range<dimensions>, size_t, Reductions ...) [with int Dim = 2; Function = main()::<lambda(hipsycl::sycl::handler&)>::<lambda(hipsycl::sycl::group<2>)>; Reductions = {}; size_t = long unsigned int]::<lambda(auto:7& ...)>&>’
/usr/include/c++/8/bits/invoke.h:89:5:   required by substitution of ‘template<class _Callable, class ... _Args> constexpr typename std::__invoke_result<_Functor, _ArgTypes>::type std::__invoke(_Callable&&, _Args&& ...) [with _Callable = hipsycl::glue::omp_dispatch::parallel_for_workgroup(Function, hipsycl::sycl::range<dimensions>, hipsycl::sycl::range<dimensions>, size_t, Reductions ...) [with int Dim = 2; Function = main()::<lambda(hipsycl::sycl::handler&)>::<lambda(hipsycl::sycl::group<2>)>; Reductions = {}; size_t = long unsigned int]::<lambda(auto:7& ...)>&; _Args = {}]’
/usr/include/c++/8/tuple:1678:27:   required from ‘constexpr decltype(auto) std::__apply_impl(_Fn&&, _Tuple&&, std::index_sequence<_Idx ...>) [with _Fn = hipsycl::glue::omp_dispatch::parallel_for_workgroup(Function, hipsycl::sycl::range<dimensions>, hipsycl::sycl::range<dimensions>, size_t, Reductions ...) [with int Dim = 2; Function = main()::<lambda(hipsycl::sycl::handler&)>::<lambda(hipsycl::sycl::group<2>)>; Reductions = {}; size_t = long unsigned int]::<lambda(auto:7& ...)>&; _Tuple = std::tuple<>&; long unsigned int ..._Idx = {}; std::index_sequence<_Idx ...> = std::integer_sequence<long unsigned int>]’
/usr/include/c++/8/tuple:1687:31:   [ skipping 10 instantiation contexts, use -ftemplate-backtrace-limit=0 to disable ]
/usr/include/c++/8/bits/invoke.h:89:5:   required by substitution of ‘template<class _Callable, class ... _Args> constexpr typename std::__invoke_result<_Functor, _ArgTypes>::type std::__invoke(_Callable&&, _Args&& ...) [with _Callable = hipsycl::sycl::handler::parallel_for_work_group(hipsycl::sycl::range<Dim>, hipsycl::sycl::range<Dim>, const ReductionsAndKernel& ...) [with KernelName = main()::<lambda(hipsycl::sycl::handler&)>::wg; ReductionsAndKernel = {main()::<lambda(hipsycl::sycl::handler&)>::<lambda(hipsycl::sycl::group<2>)>}; int dimensions = 2]::<lambda(auto:15&&, auto:16&& ...)>&; _Args = {main()::<lambda(hipsycl::sycl::handler&)>::<lambda(hipsycl::sycl::group<2>)>}]’
/usr/include/c++/8/tuple:1678:27:   required from ‘constexpr decltype(auto) std::__apply_impl(_Fn&&, _Tuple&&, std::index_sequence<_Idx ...>) [with _Fn = hipsycl::sycl::handler::parallel_for_work_group(hipsycl::sycl::range<Dim>, hipsycl::sycl::range<Dim>, const ReductionsAndKernel& ...) [with KernelName = main()::<lambda(hipsycl::sycl::handler&)>::wg; ReductionsAndKernel = {main()::<lambda(hipsycl::sycl::handler&)>::<lambda(hipsycl::sycl::group<2>)>}; int dimensions = 2]::<lambda(auto:15&&, auto:16&& ...)>&; _Tuple = std::tuple<main()::<lambda(hipsycl::sycl::handler&)>::<lambda(hipsycl::sycl::group<2>)> >; long unsigned int ..._Idx = {0}; std::index_sequence<_Idx ...> = std::integer_sequence<long unsigned int, 0>]’
/usr/include/c++/8/tuple:1687:31:   required from ‘constexpr decltype(auto) std::apply(_Fn&&, _Tuple&&) [with _Fn = hipsycl::sycl::handler::parallel_for_work_group(hipsycl::sycl::range<Dim>, hipsycl::sycl::range<Dim>, const ReductionsAndKernel& ...) [with KernelName = main()::<lambda(hipsycl::sycl::handler&)>::wg; ReductionsAndKernel = {main()::<lambda(hipsycl::sycl::handler&)>::<lambda(hipsycl::sycl::group<2>)>}; int dimensions = 2]::<lambda(auto:15&&, auto:16&& ...)>&; _Tuple = std::tuple<main()::<lambda(hipsycl::sycl::handler&)>::<lambda(hipsycl::sycl::group<2>)> >]’
/opt/hipSYCL/include/hipSYCL/sycl/detail/util.hpp:93:13:   required from ‘void hipsycl::sycl::detail::separate_last_argument_and_apply(F&&, Args&& ...) [with F = hipsycl::sycl::handler::parallel_for_work_group(hipsycl::sycl::range<Dim>, hipsycl::sycl::range<Dim>, const ReductionsAndKernel& ...) [with KernelName = main()::<lambda(hipsycl::sycl::handler&)>::wg; ReductionsAndKernel = {main()::<lambda(hipsycl::sycl::handler&)>::<lambda(hipsycl::sycl::group<2>)>}; int dimensions = 2]::<lambda(auto:15&&, auto:16&& ...)>&; Args = {const main()::<lambda(hipsycl::sycl::handler&)>::<lambda(hipsycl::sycl::group<2>)>&}]’
/opt/hipSYCL/include/hipSYCL/sycl/handler.hpp:196:45:   required from ‘void hipsycl::sycl::handler::parallel_for_work_group(hipsycl::sycl::range<Dim>, hipsycl::sycl::range<Dim>, const ReductionsAndKernel& ...) [with KernelName = main()::<lambda(hipsycl::sycl::handler&)>::wg; ReductionsAndKernel = {main()::<lambda(hipsycl::sycl::handler&)>::<lambda(hipsycl::sycl::group<2>)>}; int dimensions = 2]’
/home/park/example/example.cpp:15:14:   required from here
/opt/hipSYCL/include/hipSYCL/glue/omp/omp_kernel_launcher.hpp:295:28: error: could not convert ‘0’ from ‘int’ to ‘hipsycl::sycl::group<2>::id_type’ {aka ‘hipsycl::sycl::id<2>’}
           sycl::group<Dim> this_group{group_id, local_size, num_groups};
                            ^~~~~~~~~~
/opt/hipSYCL/include/hipSYCL/glue/omp/omp_kernel_launcher.hpp:295:28: error: could not convert ‘0’ from ‘int’ to ‘hipsycl::sycl::group<2>::id_type’ {aka ‘hipsycl::sycl::id<2>’}
/opt/hipSYCL/include/hipSYCL/glue/omp/omp_kernel_launcher.hpp:118:6: error: ‘void hipsycl::glue::omp_dispatch::iterate_range_omp_for(hipsycl::sycl::range<dimensions>, Function) [with int Dim = 2; Function = hipsycl::glue::omp_dispatch::parallel_for_workgroup(Function, hipsycl::sycl::range<dimensions>, hipsycl::sycl::range<dimensions>, size_t, Reductions ...) [with int Dim = 2; Function = main()::<lambda(hipsycl::sycl::handler&)>::<lambda(hipsycl::sycl::group<2>)>; Reductions = {}; size_t = long unsigned int]::<lambda(auto:7& ...)> [with auto:7 = {}]::<lambda(hipsycl::sycl::id<2>)>]’, declared using local type ‘hipsycl::glue::omp_dispatch::parallel_for_workgroup(Function, hipsycl::sycl::range<dimensions>, hipsycl::sycl::range<dimensions>, size_t, Reductions ...) [with int Dim = 2; Function = main()::<lambda(hipsycl::sycl::handler&)>::<lambda(hipsycl::sycl::group<2>)>; Reductions = {}; size_t = long unsigned int]::<lambda(auto:7& ...)> [with auto:7 = {}]::<lambda(hipsycl::sycl::id<2>)>’, is used but never defined [-fpermissive]
 void iterate_range_omp_for(sycl::range<Dim> r, Function f) noexcept {
      ^~~~~~~~~~~~~~~~~~~~~
```

**Describe your setup**
hipSYCL 0.9.0 (git commit cff515cc5)
clang 11, host compiler is gcc 8.4.0
used `-DHIPSYCL_TARGETS=omp` in cmake, but errors in cuda also

--- a/include/hipSYCL/sycl/libkernel/group.hpp
+++ b/include/hipSYCL/sycl/libkernel/group.hpp
@@ -441,7 +441,7 @@ struct group
         range<Dimensions> local_range,
         range<Dimensions> num_groups,
         host_barrier_type* group_barrier = nullptr,
-        id_type local_id = 0,
+        id_type local_id = {},
         void *local_memory_ptr = nullptr)
   : _group_id{group_id}, 
     _local_range{local_range}, 

